### PR TITLE
Xms Layout + Catalogs and Courses Api Calls

### PR DESCRIPTION
**Pull request by jdiouf**

### CODE REVIEW

**Review by rglasgow-deloitte — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,33 @@
-import logo from './logo.svg';
-import './App.css';
+import Header from "./components/Header/Header";
+import Footer from "./components/Footer/Footer";
+import "./App.css";
+import { Switch, Route, BrowserRouter as Router } from "react-router-dom";
+import WelcomeScreen from "./components/Welcome Screen/WelcomeScreen";
```
Comment (mattdrayer) [inline src/App.js:5]: Should there be a space in "Welcome Screen" on this line?

File: src/components/Catalogs/Catalogs.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    isLoading: false,
+    error: null,
+  });
+
+  let resultLength = 0;
+
+  const catalogs_api_url = process.env.REACT_APP_XIS_CATALOGS_API;
+  let result = (
+    <div>
+      Error loading catalogs. Please contact an administrator.
```
Comment (mattdrayer) [inline src/components/Catalogs/Catalogs.js:21]: Nit: "Please contact your administrator".  Please update all instances of this message to match.

File: src/components/Catalogs/Catalogs.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          <img src={icon} alt={"Course Catalog Image"} />
+          <div className={classes.contentWrapper} onClick={() => handleClick()}>
+            <div className={classes.catalogTitle}>{catalog}</div>
+          </div>
+        </div>
+      );
+    });
+  } else {
+    result = (
+      <div>Error Loading experiences. Please contact an administrator.</div>
```
Comment (mattdrayer) [inline src/components/Catalogs/Catalogs.js:79]: Nit:  "Loading" -> "loading"

File: src/components/Courses/Courses.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  // Initial state for tracking catalogs
+  const [coursesState, setCoursesState] = useState({
+    courses: null,
+    isLoading: false,
+    error: null
+  });
+
+  // TODO: Replace this DAU call with the selected in catalog name
+  const catalog_courses_api_url = process.env
+    .REACT_APP_XIS_COMPOSITELEDGER_API + "?provider=DAU";
```
Comment (mattdrayer) [inline src/components/Courses/Courses.js:17]: Will this TODO be addressed prior to merging the code?  If not, we should log a ticket so we don't forget about it.

File: src/components/Courses/Courses.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+  
+  let coursesLength = 0;
+
+  let result = (
+    <div>
+      Error loading courses. Please contact an administrator.
```
Comment (mattdrayer) [inline src/components/Courses/Courses.js:23]: Sometimes we say "courses", sometimes we say "experiences" -- we should be consistent in our terminology.

File: src/components/Courses/Courses.module.css [hunk 6]
```diff
@@ -0,0 +1,44 @@
+.title {
+    font-size: 3em;
+    font-weight: 500;
+    margin: 0% 1%;
+    padding-bottom: 1%;
+  }
+
+
```
Comment (mattdrayer) [inline src/components/Courses/Courses.module.css:8]: Seems like we have an extra line break or two here.

File: src/components/Courses/Courses.module.css [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+.heading{
+    font-size: 1.5em;
+    font-weight: 300;
+    border-bottom: 1em;
+}
+
+tr:nth-child(even) {
+    background-color: #dddddd;
+  }
```
Comment (mattdrayer) [inline src/components/Courses/Courses.module.css:43]: Nit: Closing bracket indentation

File: src/components/Footer/Footer.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
+import classes from "./Footer.module.css";
```
Comment (mattdrayer) [inline src/components/Footer/Footer.js:1]: We should log a ticket to make the footer a configurable/editable thing.

File: src/components/Footer/Footer.module.css [hunk 9]
```diff
@@ -0,0 +1,42 @@
+.footer {
+  color: #fff;
+  width: 100%;
+  min-height: 10px;
+  position: relative;
+  background-color: #0888aa;
+  /* background-image: linear-gradient(#2f6194 40%, #0888aa); */
```
Comment (mattdrayer) [inline src/components/Footer/Footer.module.css:7]: Should this disabled code be removed?

File: src/components/Header/Header.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import classes from "./Header.module.css";
+import { NavLink } from "react-router-dom";
+import dodLogo from "../../resources/internal/dodLogo.png";
+
+const Header = (props) => {
+  // const [showLogin, setShowLogin] = useState(false);
```
Comment (mattdrayer) [inline src/components/Header/Header.js:7]: Should this disabled code be removed?

File: src/components/Header/Header.js [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        <div className={classes.icon}>
+          <img
+            className={classes.logo}
+            src={dodLogo}
+            alt={"dodLogo"}
+            data-testid={"dod-logo"}
+          />
+          <div className={classes.logoText}>
+            <p className={classes.paraText}>
+              <b className={classes.text}>Experience Management Service</b>
```
Comment (mattdrayer) [inline src/components/Header/Header.js:58]: We should log a ticket to make this text configurable

File: src/components/Header/Header.js [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+            </p>
+            <p className={classes.paraText}>
+              <b className={classes.subText}>U.S. Department of Defense</b>
```
Comment (mattdrayer) [inline src/components/Header/Header.js:61]: We should log a ticket to make this text configurable

File: src/components/Header/Header.module.css [hunk 13]
```diff
@@ -0,0 +1,105 @@
+.imgBar {
+    max-width: 100%;
+    padding: 1% 2% 0.5% 10%;
+
+    display: flex;
+    justify-content: space-between;
+
+    background-image: url("../../resources/internal/dodBanner.png");
```
Comment (mattdrayer) [inline src/components/Header/Header.module.css:8]: We should log a ticket to make this image configurable

File: src/components/Header/Header.module.css [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+.navItemWrapperActive {
+    background-color: #337ab7;
+    display: inline-block;
+}
+
+.navItem {
+    padding: 10px 10px 10px 10px;
+    /* margin: 10px; */
```
Comment (mattdrayer) [inline src/components/Header/Header.module.css:100]: Remove disabled code prior to merging

File: src/components/Header/Header.module.css [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+    display: inline-block;
+    text-decoration: none;
+    color: #fff;
+    /* height: 50%; */
```
Comment (mattdrayer) [inline src/components/Header/Header.module.css:104]: Remove disabled code prior to merging

File: src/components/Login/Login.js [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+import { useState } from "react";
+
+const Login = (props) => {
+  let [credentials, setCredentials] = useState({
+    username: "",
+    password: "",
+  });
+
+  const forgotPassword = () => {
+    alert("Forgot Password!");
```
Comment (mattdrayer) [inline src/components/Login/Login.js:11]: We should log a ticket to implement the Forgot Password workflow

File: src/components/Courses/Courses.js (hunk 5 above)
Comment (rglasgow-deloitte) [reply src/components/Courses/Courses.js:23]: Let's go with courses.

**Review by mattdrayer — APPROVED** [review]

File: src/components/Courses/Courses.js (hunk 4 above)
Comment (rglasgow-deloitte) [reply src/components/Courses/Courses.js:17]: https://jira.deloitte.com/browse/ECC-634
We will need this until we are done with the XMS Cleanup

**Pull request merged**